receiver/adguard_poller.py

```
@@ -0,0 +1,514 @@
+
+# ── Module-level helpers ──────────────────────────────────────────────────────
+
+def _parse_ts(ts: str) -> tuple[datetime, int] | None:
+    """Parse an RFC3339/RFC3339Nano timestamp to a ``(datetime, nanoseconds)`` pair.
+
+    AdGuard Home emits nanosecond-precision timestamps, e.g.::
+
+        2026-04-13T18:10:52.123456789+02:00
+
+    Returns a ``(aware_datetime, ns_int)`` tuple where ``ns_int`` is the full
+    fractional second expressed as nanoseconds (0–999 999 999).  Using the
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)